### PR TITLE
Add DAO Test cases for throttle policies and applications

### PR DESCRIPTION
**Pull request by msm1992**
## Purpose
This PR adds DAO Tests for throttle policies and applications.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: common-java-libs/org.wso2.apk.apimgt.impl/build.gradle [hunk 1]
```diff
@@ -61,4 +65,16 @@ dependencies {
     testImplementation libs.powermock.api.mockito2
     testImplementation libs.jacoco.agent
     testImplementation libs.xml.apis
+    testImplementation group: 'org.testng', name: 'testng', version: '7.6.1'
```
Comment (RakhithaRR) [inline common-java-libs/org.wso2.apk.apimgt.impl/build.gradle:68]: Shall we add this to `lib.versions.toml`?

File: common-java-libs/org.wso2.apk.apimgt.impl/build.gradle [hunk 2]
```diff
@@ -15,6 +15,10 @@
  *
  */
 
+plugins {
+    id 'com.avast.gradle.docker-compose' version '0.16.8'
```
Comment (RakhithaRR) [inline common-java-libs/org.wso2.apk.apimgt.impl/build.gradle:19]: Shall we add this to `lib.versions.toml`?

Comment (binodmx) [reply common-java-libs/org.wso2.apk.apimgt.impl/build.gradle:19]: I think it's better to wrap docker-compose commands in `common-gradle-scripts/docker.gradle` and use here. @RakhithaRR WDYT?

Comment (RakhithaRR) [reply common-java-libs/org.wso2.apk.apimgt.impl/build.gradle]: This docker-compose plugin is used only for testing in java right? `docker.gradle` contains tasks that are used to build and push the images. So I'm not sure whether its correct to add this plugin there.

Comment (msm1992) [reply common-java-libs/org.wso2.apk.apimgt.impl/build.gradle]: Yes this plugin is only used for testing

**Pull request closed without merging**